repos/keg-components/src/components/form/checkbox/checkbox.wrapper.js

```diff
@@ -82,27 +107,84 @@ const ChildrenComponent = ({ children, className }) => (
+    ref,
+    () => ({
+      // allows consumer to fetch the latest checked value
+      isChecked,
+
+      // allows consumer to toggle the checked value, which will
+      // also fire the pressHandler
+      setChecked: checked => {
+        setChecked(checked)
+        pressHandler(checked)
+      },
+    }),
+    [ ref, isChecked, setChecked, pressHandler ]
+  )
+}
+
 /**
  * CheckboxWrapper
  * Wraps the Passed in Checkbox Element
  * @param {Object} props - see PropTypes below
- *
+ * @param {string?} props.className - css class string
+ * @param {boolean?} props.initChecked - the initial checked/unchecked value of Checkbox
+ * @param {boolean?} props.checked - alias for initChecked
+ * @param {Object | String | Array} props.children - react children
+ * @param {string} props.elType - type of checkbox
+ * @param {Component} props.Element - the which component used for the core checkbox ui. @see checkbox.js
+ * @param {Component?} props.CheckIcon - an optional component that overrides the icon used in the checkbox
+ * @param {boolean} props.disabled - disables checking and unchecking if true
+ * @param {boolean} props.disableCheck - disables checking if true
+ * @param {boolean} props.disableUncheck - disables unchecking if true
+ * @param {boolean} props.allowAdjacentPress - if true, allows pressing a side component to toggle the checkbox
+ * @param {boolean} props.isWeb - indicates current platform is web. Provide by checkbox.js
+ * @param {(Component | string)?} props.LeftComponent - an optional component or string to render left of the checkbox
+ * @param {string?} props.leftClassName - optional class name for the left component
+ * @param {boolean} props.close - if true, side components will be rendered close to the checkbox. If false, spaced apart.
+ * @param {Function?} props.onChange - callback fired when checkbox is toggled
+ * @param {Function?} props.onValueChange - callback fired when checkbox is toggled
+ * @param {Function?} props.setCheckedSetter - callback called once on initial render. Is passed the `setChecked` callback to provide consumer control of checked/unchecked state if needed.
+ * @param {(Component | string)?} props.RightComponent -  an optional component or string to render right of the checkbox
+ * @param {string?} props.rightClassName - optional class name for the right component
+ * @param {Object} props.styles - styles that overwrite default checkbox styles
+ * @param {Component?} props.CheckboxComponent - an optional component that overrides the default checkbox element, passed the same props and styles
+ * @param {string} props.type - theme type
+ * @param {string} props.themePath - path to theme object
+ * @param {boolean} props.value - alias for initChecked
+ * @param {...rest} props.* - the remaining props are passed to Element
  */
-export const CheckboxWrapper = props => {
+export const CheckboxWrapper = forwardRef((props, ref) => {
   const {
     className,
+    initChecked,
```

Comment:
lol... that's a lot of props just for a checkbox

Reply:
Yeah, it's doing a lot, but on the other hand, it's super versatile